lib/modules/dosomething/dosomething_user/dosomething_user_social_auth.inc

```diff
@@ -0,0 +1,122 @@
+<?php
+
+/**
+ * Constructs a new Drupal user for the given account details to represent a Facebook user.
+ * @param  array $account_details  All of the properties to instantiate
+ * @return $user
+ */
+function dosomething_user_make_facebook_user($account_details) {
+  $edit = [
+    'mail' => $account_details['email'],
+    'name' => $account_details['email'],
+    'pass' => user_password(),
+    'status' => 1,
+    'created' => REQUEST_TIME,
+  ];
+
+  dosomething_user_set_fields($edit, $account_details);
+
+  $user = user_save(null, $edit);
+  user_save($user, ['name' => $user->uid]);
+
+  return $user;
+}
+
+/**
+ * Registers a new user with Northstar (or upserts with existing) & returns a Drupal account
+ * @param  array $account_details Account details for the user to register
+ * @return $user Created (or retrieved) Drupal user
```

Comment:
The [`@return` annotation](https://www.phpdoc.org/docs/latest/references/phpdoc/tags/return.html) should have a type, not a name.